openaerostruct/docs/advanced_features/scripts/basic_2_sec.py

```diff
@@ -0,0 +1,237 @@
+# In this example, we only wish to constraint the x-distance between the sections at both the leading and trailing edge.
+
+multi_geom_group = MultiSecGeometry(
+    surface=surface, joining_comp=True, dim_constr=[np.array([1, 0, 0]), np.array([1, 0, 0])]
+)
+prob.model.add_subsystem(surface["name"], multi_geom_group)
+
+# docs checkpoint 4
+
+# In this next part, we will setup the aerodynamics group. First we use a utility function called build_sections which takes our multi-section surface dictionary and outputs a
+# surface dictionary for each individual section. We then inputs these dictionaries into the mesh unification function unify_mesh to produce a single mesh array for the the entire surface.
+# We then add this mesh to the multi-section surface dictionary
+section_surfaces = build_sections(surface)
+uniMesh = unify_mesh(section_surfaces)
+surface["mesh"] = uniMesh
+
+# Create the aero point group, which contains the actual aerodynamic
+# analyses. This step is exactly as it's normally done except the surface dictionary we pass in is the multi-surface one
+aero_group = AeroPoint(surfaces=[surface])
+point_name = "aero_point_0"
+prob.model.add_subsystem(point_name, aero_group, promotes_inputs=["v", "alpha", "Mach_number", "re", "rho", "cg"])
+
+# docs checkpoint 5
+
+# The following steps are similar to a normal OAS surface script but note the differences.
+
+# Get name of surface and construct the name of the unified surface mesh
+name = surface["name"]
+unification_name = "{}_unification".format(surface["name"])
+
+# Connect the mesh from the mesh unification component to the analysis point.
+prob.model.connect(name + "." + unification_name + "." + name + "_uni_mesh", point_name + "." + "surface" + ".def_mesh")
+
+# Perform the connections with the modified names within the
+# 'aero_states' group.
+prob.model.connect(
+    name + "." + unification_name + "." + name + "_uni_mesh", point_name + ".aero_states." + "surface" + "_def_mesh"
+)
+
+# docs checkpoint 6
+
+# Next, we add the DVs to the OpenMDAO problem. Note that each surface's geometeric parameters are under the given section names specified in the multi-surface dictionary earlier.
+# Here we use the chord B-spline that we specified earlier for each section and the angle-of-attack as DVs.
+prob.model.add_design_var("surface.sec0.chord_cp", lower=0.1, upper=10.0, units=None)
+prob.model.add_design_var("surface.sec1.chord_cp", lower=0.1, upper=10.0, units=None)
+prob.model.add_design_var("alpha", lower=0.0, upper=10.0, units="deg")
+
+
+# Next, we add the C0 continuity constraint for this problem by constraining the x-distance between sections to 0.
+# NOTE: SLSQP optimizer does not handle this equality constraint properly so the constraint needs to be specified as an inequality constraint
+# All other optimizers like SNOPT can handle the equality constraint as is.
+
+prob.model.add_constraint("surface.surface_joining.section_separation", upper=0, lower=0)  # FOR SLSQP
```

Comment:
I would prefer to leave it as is. It seems like SLSQP sometimes requires the scaling on this constraint to be changed in order to converge properly or may require that it be formulated as an inequality constraint. Since I'm not sure what optimizer the user will be using I prefer to leave it this way.